src/main/kotlin/blackjack/domain/Participants.kt

```diff
@@ -0,0 +1,50 @@
+package blackjack.domain
+
+class Participants(private val participants: List<Participant>) {
+    val elements: List<Participant>
+        get() = participants.toList()
+
+    init {
+        require(participants.isNotEmpty()) { "참여자는 비어있을 수 없습니다." }
+        requiresOnlyOneDealer()
+        requiresAtLeastOneGambler()
+    }
+
+    private fun requiresOnlyOneDealer() {
+        val dealers = participants.filterIsInstance<Dealer>()
+        require(dealers.size == 1) { "게임 내 딜러는 한명만 참여 가능합니다." }
+    }
+
+    private fun requiresAtLeastOneGambler() {
+        val gamblers = extractGamblers()
+        require(gamblers.isNotEmpty()) { "게임 내 겜블러는 한명 이상 참여해야 합니다." }
+    }
+
+    fun receiveTwoCardsEach(deck: Deck) {
+        participants.forEach { participant ->
+            participant.receive(deck.draw(), deck.draw())
+        }
+    }
+
+    fun extractGamblers(): List<Gambler> {
+        return participants.filterIsInstance<Gambler>()
+    }
+
+    fun extractNames(): List<String> {
+        return participants.map { participant -> participant.name }
+    }
+
+    fun extractDealer(): Dealer {
+        return participants.filterIsInstance<Dealer>().first()
+    }
+
+    companion object {
+        fun of(
+            dealer: Dealer,
+            gamblers: List<Gambler>,
+        ): Participants {
+            val participants = listOf(dealer) + gamblers
```

Comment:
꼭 dealer를 list로 변경하지 않아도 될 것 같아요. 😃
```suggestion
            val participants = gamblers + dealer
```

Reply:
딜러가 맨 앞에 있게 고정하려다보니, 다소 이런 억지스러운 코드가 나왔습니다 😓
그런데, 생각해보니 딜러가 맨처음에 있어야 하는 것은 view에서 출력할 때 밖에 없어서, 그 부분은 view에서 담당하는 것이 좋겠다는 생각이 드네요! 변경하였습니다~!

Reply:
아~ 그런 의도가 있으셨군요?  
일단 그런 의도라면 확장 함수를 만들어 보시는 것도 괜찮을 것 같아요. 

Collection의 plus 확장 함수가 어떻게 구현되어있는지 참고해서   `Participant.plus` 혹은 `T.plus` operator를 재정의해보시면 어떨까요? 😃

Reply:
아~ 그런 의도가 있으셨군요?  
일단 그런 의도라면 확장 함수를 만들어 보시는 것도 괜찮을 것 같아요. 

Collection의 plus 확장 함수가 어떻게 구현되어있는지 참고해서   `Participant.plus` 혹은 `T.plus` operator를 재정의해보시면 어떨까요? 😃